### PR TITLE
Refactor generator

### LINKED ISSUE #10: Refactor generator.py
The code has room for improvement and does all in one place. In my opinion this needs to be refactored to come in smaller portions such that also tests can be written for it

### PR DESCRIPTION
**Pull request by schaefi**
This change is many fold please review on a commit base

**Refactor generator**
    
Turn the generator into a class. Apply the single responsibilty OO pattern. Add tests for the code and give the single parts of the processing a name. Update the keg commandline code to make use of the classes. Keg should also be usuable as an API,
thus I think it's important come up with a good class design. This Fixes #10

**Added anymarkup to virtual environment**
    
keg will also support YAML output. For this the anymarkup  module is needed.

**Added support for YAML markup in KiwiDescription**

### CODE REVIEW

**Comment by schaefi** [comment]
@jesusbv if you review the Generator I suggest you look at the complete code in generator.py. I moved this into a class and also changed the design such that this code can be used easier in an API and is from my pov easier to read and to understand. That causes the diff to be hard to read. Thus just take a look at the complete file. Thanks

File: keg/generator.py [hunk 1]
```diff
@@ -19,69 +19,92 @@
...
-        else:
-            raise KegError(errmsg)
+    def __init__(
+        self, image_definition: KegImageDefinition, dest_dir: str
+    ):
+        self.image_definition = image_definition
+        self.description_schemas = os.path.join(
+            image_definition.recipes_root, 'schemas'
+        )
+        self.dest_dir = dest_dir
```
Comment (rjschwei) [inline keg/generator.py:46]: IMHO that dest_dir is in fact a directory should be checked 

if no os.path.isdir(dest_dir):
  COMPLAIN

Comment (schaefi) [reply keg/generator.py:52]: yes fully agree and done

File: keg/generator.py [hunk 2]
```diff
@@ -19,69 +19,98 @@
...
+        kiwi_document = kiwi_template.render(
+            data=self.image_definition.data
+        )
+        with open(outfile, 'w') as kiwi_config:
+            kiwi_config.write(kiwi_document)
+        kiwi = KiwiDescription(outfile)
+        if markup == 'xml':
+            kiwi.create_XML_description(outfile)
+        if markup == 'yaml':
+            kiwi.create_YAML_description(outfile)
```
Comment (jesusbv) [inline keg/generator.py:84]: I think we  should add an `else:` option to log | print a message informing that that markup is not supported.

Comment (schaefi) [reply keg/generator.py:84]: yes I think it should raise an error in this case

Comment (schaefi) [reply keg/generator.py:84]: done

**Review by jesusbv — APPROVED** [review]
LGTM

**Pull request merged**